sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -69,6 +69,21 @@ std::string getDeviceTypeName(const device &Device) {
   }
 }
 
+const char *getArchName(const device &Device) {
+  namespace syclex = sycl::ext::oneapi::experimental;
+  auto arch = Device.get_info<syclex::info::device::architecture>();
+  switch (arch) {
+#define __SYCL_ARCHITECTURE(ARCH, VAL)                                         \
+  case syclex::architecture::ARCH:                                             \
+    return #ARCH;
+#define __SYCL_ARCHITECTURE_ALIAS(ARCH, VAL)
+#include <sycl/ext/oneapi/experimental/architectures.def>
+#undef __SYCL_ARCHITECTURE
+#undef __SYCL_ARCHITECTURE_ALIAS
+  }
+  assert(false && "Unreachable!");
```

Comment:
We now have `unknown` architecture. I suppose that it should technically be a `default:` clause within the switch, but it is hard to achieve, so I suggest that we just return `unknown` here to avoid warnings that you see on Windows